makefiles/Makefile.python.mk

```diff
@@ -377,104 +508,206 @@ else
 	cd temp && tar -c -v -z --no-same-owner -f ../or-tools_python_examples$(PYPI_OS)_v$(OR_TOOLS_VERSION).tar.gz ortools_examples
 endif
 
+#####################
+##  Pypi artifact  ##
+#####################
 PYPI_ARCHIVE_TEMP_DIR = temp-python$(PYTHON_VERSION)
 
-OR_TOOLS_PYTHON_GEN_SCRIPTS = $(wildcard src/gen/ortools/*/*.py) $(wildcard src/gen/ortools/*/*.cc)
+MISSING_PYPI_FILES = \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/setup.py \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/README.txt \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/LICENSE-2.0.txt \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/__init__.py \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/algorithms \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/graph \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/constraint_solver \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/linear_solver \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/sat \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/data \
+ $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools/util \
+
+$(PYPI_ARCHIVE_TEMP_DIR):
+	$(MKDIR) $(PYPI_ARCHIVE_TEMP_DIR)
+
+$(PYPI_ARCHIVE_TEMP_DIR)/ortools: | $(PYPI_ARCHIVE_TEMP_DIR)
+	$(MKDIR) $(PYPI_ARCHIVE_TEMP_DIR)$Sortools
 
-# Stages all the files needed to build the python package.
-pypi_archive_dir: python $(PYPI_ARCHIVE_TEMP_DIR)
+$(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools: | $(PYPI_ARCHIVE_TEMP_DIR)/ortools
+	$(MKDIR) $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$Sortools
 
-# Patches the archive files to be able to build a pypi package.
-# Graft libortools if needed and set RPATHs.
-pypi_archive: pypi_archive_dir $(PATCHELF)
-ifneq ($(SYSTEM),win)
-	cp lib/libortools.$(LIB_SUFFIX) $(PYPI_ARCHIVE_TEMP_DIR)/ortools/ortools
-ifeq ($(PLATFORM),MACOSX)
-	tools/fix_python_libraries_on_mac.sh $(PYPI_ARCHIVE_TEMP_DIR)
+$(PYPI_ARCHIVE_TEMP_DIR)/ortools/README.txt: tools/README.pypi | $(PYPI_ARCHIVE_TEMP_DIR)/ortools
+	$(COPY) tools$SREADME.pypi $(PYPI_ARCHIVE_TEMP_DIR)$Sortools$SREADME.txt
+
+$(PYPI_ARCHIVE_TEMP_DIR)/ortools/LICENSE-2.0.txt: LICENSE-2.0.txt | $(PYPI_ARCHIVE_TEMP_DIR)/ortools
+	$(COPY) LICENSE-2.0.txt $(PYPI_ARCHIVE_TEMP_DIR)$Sortools
+
+PYTHON_SETUP_DEPS=
+ifeq ($(UNIX_GFLAGS_DIR),$(OR_TOOLS_TOP)/dependencies/install)
+  ifeq ($(PLATFORM),MACOSX)
+    PYTHON_SETUP_DEPS += , 'libgflags.2.2.$L'
```

Comment:
Should be $(PYTHON_SWIG_SUFFIX) on MacOS i.e. python look for .so even on Mac -_-

Reply:
it's ok this is dynamic dependencies not pywrappers so on MacOS they are suffixed by .dylib